device.go

```diff
@@ -32,6 +32,7 @@ type device struct {
 	BootEnvReadWriter
 	Commander
 	*partitions
+	hasUpdateCb func(d *device) (bool, error)
```

Comment:
An example of injecting `BootEnvReadWriter` with desired data is right here: https://github.com/mendersoftware/mender/pull/218/commits/f3e8bea7137b061c08ebc6c95cf30c0dba94e499#diff-eed9e366efcb961973c3dbd3d4b7d620R172